File: Calculator/app/src/main/java/com/example/calculator/MainActivity.java
package com.example.calculator;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }


    public void plus(View v){
        EditText num1 = (EditText) findViewById(R.id.num1);
        EditText num2 = (EditText) findViewById(R.id.num2);
        EditText result = (EditText) findViewById(R.id.result);
        Integer num3 = Integer.parseInt(num1.getText().toString()) +Integer.parseInt(num2.getText().toString());
        result.setText(num3.toString());

    }
    public void minus(View v){
        EditText num1 = (EditText) findViewById(R.id.num1);
        EditText num2 = (EditText) findViewById(R.id.num2);
        EditText result = (EditText) findViewById(R.id.result);
        Integer num3 = Integer.parseInt(num1.getText().toString()) -Integer.parseInt(num2.getText().toString());
        result.setText(num3.toString());
    }
    public void multiply(View v){
        EditText num1 = (EditText) findViewById(R.id.num1);
        EditText num2 = (EditText) findViewById(R.id.num2);
        EditText result = (EditText) findViewById(R.id.result);
        Integer num3 = Integer.parseInt(num1.getText().toString()) *Integer.parseInt(num2.getText().toString());
        result.setText(num3.toString());
    }
    public void division(View v){
        EditText num1 = (EditText) findViewById(R.id.num1);
        EditText num2 = (EditText) findViewById(R.id.num2);
        EditText result = (EditText) findViewById(R.id.result);
        Integer num3 = Integer.parseInt(num1.getText().toString()) / Integer.parseInt(num2.getText().toString());
        result.setText(num3.toString());
    }
}

File: KTFCMpushnotice/app/src/main/java/com/example/ktfcmpushnotice/MainActivity.kt
package com.example.ktfcmpushnotice

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import com.google.firebase.messaging.FirebaseMessaging

class MainActivity : AppCompatActivity() {
    //이게 게으른 인스턴스화
    private val resultTextView: TextView by lazy{
        findViewById(R.id.resultTextView)
    }
    private val firebaseToken: TextView by lazy{
        findViewById(R.id.firebaseTokenText)
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initFirebase()
    }
    private fun initFirebase(){
        FirebaseMessaging.getInstance().token
            .addOnCompleteListener(){ task ->
                if(task.isSuccessful){
                    firebaseToken.text = task.result;
                }
            }
    }
}